src/app/CommandHandler.h

```diff
@@ -85,6 +85,12 @@ class CommandHandler
          * fails to exist.
          */
         virtual Protocols::InteractionModel::Status CommandExists(const ConcreteCommandPath & aCommandPath) = 0;
+
+        /*
+         * Get the magic number of the InteractionModelEngine. A CommandHandler::Handle is valid iff
+         * its magic number is equals to the InteractionModelEngine's one.
+         */
+        virtual uint32_t GetInteractionModelEngineMagicNumber() const = 0;
```

Comment:
This needs better naming.  While it was all internal API this "magic number" business was fine, maybe, but if this is going to be public API we should have a better name.

Reply:
Do you have any suggestion? What about `GetCommandHandlerValidationNumber`?

Reply:
When does this number update?  Is this basically GetInteractionModelEngineGeneration()?

Reply:
Yes, it is. It is updated during InteractionModelEngine init() and shutdown()

Reply:
Then maybe we should call it that...